src/charm.py

```diff
@@ -319,6 +329,10 @@ def _update_ingress(self) -> None:
             ingress_controller = self._get_ingress_controller(namespace)
             ingress_ips = ingress_controller.get_ingress_ips()
             message = f"Ingress IP(s): {', '.join(ingress_ips)}" if ingress_ips else ""
+
+            if self.model.get_relation("ingress") is not None:
+                self._ingress_provider.publish_url(relation, ("http://" + hostnames[0]))
```

Comment:
Will `hostnames` always contain an element at this point?

Reply:
We should not hardcode `http` here, if the `tls-certificates` relation is set, we should be in `https`

Reply:
Oh and the URL may contain a path too... definitely the way the URL is build should be better though !